whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotFileGenerator.java

```diff
@@ -36,88 +39,66 @@ public class SnapshotFileGenerator {
+    public void createSnapshots() {
+        final Stopwatch stopwatch = Stopwatch.createStarted();
+        final List<NrtmSource> sources = getSources();
         final List<NrtmVersionInfo> sourceVersions = nrtmVersionInfoRepository.findLastVersionPerSource();
-        if (sourceVersions.isEmpty()) {
-            LOGGER.info("Initializing NRTM");
-            for (final NrtmSourceModel source : sourceRepository.getSources()) {
-                final NrtmVersionInfo version = nrtmVersionInfoRepository.createInitialVersion(source, state.serialId());
-                sourceVersions.add(version);
-            }
-        } else {
-            if (!snapshotGenerationWindow.isInWindow()) {
-                return;
-            }
-            sourceVersions.removeIf(versionToRemove ->
-                versionToRemove.type() == NrtmDocumentType.SNAPSHOT);
-            if (sourceVersions.isEmpty()) {
-                return;
-            }
-            sourceVersions.removeIf(versionToRemove -> !snapshotGenerationWindow.hasVersionExpired(
-                nrtmVersionInfoRepository.findLastSnapshotVersionForSource(
-                    versionToRemove.source()
-                ))
-            );
-            if (sourceVersions.isEmpty()) {
-                return;
-            }
-        }
-        createSnapshotsForVersions(state, sourceVersions.stream().map(nrtmVersionInfoRepository::saveNewSnapshotVersion).toList());
-    }
 
-    public List<NrtmVersionInfo> createInitialSnapshots(final SnapshotState state) {
-        sourceRepository.createSources();
-        final List<NrtmVersionInfo> sourceVersions = new ArrayList<>();
-        for (final NrtmSourceModel source : sourceRepository.getSources()) {
-            final NrtmVersionInfo version = nrtmVersionInfoRepository.createInitialVersion(source, state.serialId());
-            sourceVersions.add(version);
-        }
-        return createSnapshotsForVersions(state, sourceVersions);
-    }
+        final SnapshotState snapshotState = whoisObjectRepository.getSnapshotState(sourceVersions.isEmpty() ? null : sourceVersions.get(0).lastSerialId());
+        LOGGER.info("Found {} objects in {}", snapshotState.whoisObjectData().size(), stopwatch);
 
-    private List<NrtmVersionInfo> createSnapshotsForVersions(final SnapshotState state, final List<NrtmVersionInfo> snapshotVersions) {
-        final Stopwatch stopwatch = Stopwatch.createStarted();
         final List<Thread> queueReaders = new ArrayList<>();
         final Map<CIString, LinkedBlockingQueue<RpslObjectData>> queueMap = new HashMap<>();
-        for (final NrtmVersionInfo snapshotVersion : snapshotVersions) {
-            LOGGER.info("Creating snapshot for {}", snapshotVersion.source().getName());
-            final PublishableNrtmFile snapshotFile = new PublishableSnapshotFile(snapshotVersion);
+        for (final NrtmSource source : sources) {
+
+            if(!canProceed(sourceVersions, source)) {
```

Comment:
Just for my understanding: Check if there are changes between snapshots, if there are not changes then we should generate new snapshot

Reply:
RFC "The mirror server MUST generate a new Snapshot File between once
      per hour and once per day, if there have been changes to the IRR
      objects." so if there are no deltas since last snapshot created it will not ceate new snapshot